webcompat/api/uploads.py

```diff
@@ -26,11 +23,71 @@
 uploads = Blueprint('uploads', __name__, url_prefix='/upload',
                     template_folder='../templates')
 JSON_MIME = 'application/json'
-images = UploadSet('uploads', IMAGES)
-configure_uploads(app, images)
+ALLOWED_FORMATS = ('jpg', 'jpeg', 'jpe', 'png', 'gif', 'bmp')
 
-# limit image uploads to 4MB
-patch_request_class(app, 4 * 1024 * 1024)
+
+class UploadNotAllowed(Exception):
+    """
+    This exception is raised when we try to save a file that is not
+    allowed. Allowed files are defined in ALLOWED_FORMATS
+    """
+
+
+class Upload(object):
+    '''Class that abstracts over saving image and screenshot uploads.
+
+    It performs a simple extension-based validation before saving to the
+    file system. If a file is not allowed, a UploadNotAllowed exception
+    is rased.
+    '''
+    def __init__(self, imagedata):
+        self.imagedata = imagedata
+        self.is_base_64 = (isinstance(imagedata, unicode) and
+                           imagedata.startswith('data:image/'))
+        self.is_filestorage = isinstance(imagedata, FileStorage)
+        self.file_ext = self.get_file_ext(imagedata)
+
+    def get_file_ext(self, imagedata):
+        '''Method to return the file extension.'''
+        if self.is_filestorage:
+            return imagedata.filename.split('.')[-1].lower()
+        if self.is_base_64:
+            # We only expect base64 png screenshot uploads,
+            # so we do a double check to verify that, and raise otherwise
+            # data:image/png;base64, <base64 string>
+            if imagedata.split('/')[1].split(';')[0] != 'png':
+                raise UploadNotAllowed()
+            return 'png'
```

Comment:
Also Pillow provides specific way to test the right file format instead of the extension which is a ok test for validity but not a real one. 

```python
>>> from PIL import Image
>>> im = Image.open('pacifique-fuji.jpg')
>>> im.format
'JPEG'
>>> 
```

Also when the file is not correct, we get an error.

``` python
>>> im2 = Image.open('foobar.pdf')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/karl/.virtualenvs/webcompatcom/lib/python2.7/site-packages/PIL/Image.py", line 2286, in open
    % (filename if filename else fp))
IOError: cannot identify image file 'foobar.pdf'
```